packages/gitbook/src/routes/llms.ts

```diff
@@ -140,46 +141,68 @@ async function getNodesFromSiteSpaces(
-                    const children: Paragraph['children'] = [
-                        {
-                            type: 'link',
-                            url,
-                            children: [{ type: 'text', value: page.title }],
-                        },
-                    ];
-                    if (page.description) {
-                        children.push({ type: 'text', value: `: ${page.description}` });
-                    }
-                    return {
-                        type: 'listItem',
-                        children: [{ type: 'paragraph', children }],
-                    };
-                })
-            );
-            const nodes: RootContent[] = [];
-            if (options.heading) {
-                nodes.push({
-                    type: 'heading',
-                    depth: 2,
-                    children: [{ type: 'text', value: siteSpace.title }],
-                });
-            }
-            nodes.push({
-                type: 'list',
-                spread: false,
-                children: listChildren,
+            return getMarkdownForPagesTree(pages, {
+                siteSpaceUrl,
+                linker,
+                heading: options.heading ? siteSpace.title : undefined,
+                withMarkdownPages: options.withMarkdownPages,
             });
-            return nodes;
         })
     );
     return all.flat();
 }
+
+/**
+ * Returns a list of markdown nodes for a pages tree.
+ */
+export async function getMarkdownForPagesTree(
+    pages: FlatPageEntry[],
+    options: {
+        siteSpaceUrl: string;
+        linker: GitBookLinker;
+        heading?: string;
+        withMarkdownPages?: boolean;
+    }
+): Promise<RootContent[]> {
+    const { siteSpaceUrl, linker } = options;
+
+    const listChildren = await Promise.all(
+        pages.map(async ({ page }): Promise<ListItem> => {
+            const pageURL = new URL(siteSpaceUrl);
+            pageURL.pathname = joinPath(pageURL.pathname, page.path);
+            if (options.withMarkdownPages) {
+                pageURL.pathname = `${pageURL.pathname}.md`;
+            }
+
+            const url = linker.toLinkForContent(pageURL.toString());
+            const children: Paragraph['children'] = [
+                {
+                    type: 'link',
+                    url,
+                    children: [{ type: 'text', value: page.title }],
+                },
+            ];
+            if (page.description) {
+                children.push({ type: 'text', value: `: ${page.description}` });
+            }
+            return {
+                type: 'listItem',
+                children: [{ type: 'paragraph', children }],
+            };
+        })
+    );
+    const nodes: RootContent[] = [];
+    if (options.heading) {
+        nodes.push({
+            type: 'heading',
+            depth: 2,
+            children: [{ type: 'text', value: options.heading }],
+        });
+    }
```

Comment:
This doesn't need to be done in this function; the place that calls this function can just put the heading before calling the function

Better to limit the scope of each low-level function